service/lib/dinstaller/dbus/storage/actions.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2022] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "dbus"
+require "dinstaller/storage/actions"
+
+module DInstaller
+  module DBus
+    module Storage
+      # D-Bus object to get the list of actions to perform in the system
+      class Actions < ::DBus::Object
+        PATH = "/org/opensuse/DInstaller/Storage/Actions1"
+        private_constant :PATH
+
+        INTERFACE = "org.opensuse.DInstaller.Storage.Actions1"
+        private_constant :INTERFACE
+
+        def initialize(logger)
+          @logger = logger
+
+          super(PATH)
+        end
+
+        dbus_interface INTERFACE do
+          # TODO: emit a signal when actions change (e.g., when a proposal is calculated). Right
```

Comment:
another easy way without subscribing is to pass this dbus object to Proposal and call signal emit there.

Reply:
Shouldn't it be similar to implementing the observer pattern? Anyway, I would do nothing for now. As we agreed, we need to take a decision about the general approach to follow.